eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringMongoReader.java

```diff
@@ -85,11 +85,20 @@ public void open(ExecutionContext executionContext) throws ItemStreamException {
     public void initializeReader() {
         Bson query = Filters.and(Filters.in(ASSEMBLY_FIELD, assembly),
                                  Filters.exists(CLUSTERED_VARIANT_ACCESSION_FIELD, readOnlyClusteredVariants));
+        // When clustering variant that do not have any RSID we do not want to retrieve any dbSNP submitted variants
+        // We use this filters that should never return any results, for this purpose
+        Bson query_no_results = Filters.eq("_id", "-1");
```

Comment:
I understand this works but doing a query that we know will always return nothing seems weird.
What about setting the `dbsnpCursor` to null so it is clear that there is nothing to be read in that case.
```
        if (readOnlyClusteredVariants) {
            FindIterable<Document> submittedVariantsDbsnp =
                    getSubmittedVariants(query, DbsnpSubmittedVariantEntity.class);
            dbsnpCursor = submittedVariantsDbsnp.iterator();
        } else {
            dbsnpCursor = null;
        }
```
And in the `read` method validate that 
```
        Document nextElement = (dbsnpCursor != null && dbsnpCursor.hasNext()) ? dbsnpCursor.next() :
                (evaCursor.hasNext() ? evaCursor.next() : null);
```
And this would involve checking it is not null in the `close` method

Reply:
Yes, that is much better.